### ISSUE TITLE
[MDEP-49] Can't unpack multiple classifiers from same artifact; "already unpacked"

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-dependency-plugin#620]
**[duality72](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=duality72)** opened **[MDEP-49](https://issues.apache.org/jira/browse/MDEP-49?redirect=false)** and commented

A plugin configuration like this does not work:

    <plugin>      
      <groupId>org.codehaus.mojo</groupId>
      <artifactId>dependency-maven-plugin</artifactId>
      <executions>
        <execution>
          <id>generate-resources-apphome</id>
          <phase>generate-resources</phase>
          <configuration>
            <artifactItems>
              <artifactItem>
                <groupId>com.example</groupId>
                <artifactId>MyProject</artifactId>
                <version>5.0-SNAPSHOT</version>
                <classifier>someFiles</classifier>
                <type>zip</type>
                <outputDirectory>target/dir1</outputDirectory>
              </artifactItem>
              <artifactItem>
                <groupId>com.example</groupId>
                <artifactId>MyProject</artifactId>
                <version>5.0-SNAPSHOT</version>
                <classifier>otherFiles</classifier>
                <type>zip</type>
                <outputDirectory>target/dir2</outputDirectory>
              </artifactItem>
            </artifactItems>
          </configuration>
          <goals>
            <goal>unpack</goal>
          </goals>
        </execution>
      </executions>
    </plugin>

The otherFiles.zip file is never unpacked to dir2 and the conole output states:

    [INFO] MyProject-5.0-SNAPSHOT-otherFiles.zip already unpacked.

When actually it was the someFiles.zip that was already unpacked.

---

**Affects:** 1.0

**Comment by jira-importer** [comment]
**[Brian E. Fox (imported)](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brianf@infinity.nu)** commented

Can you try this with org.apache.maven.plugins:maven-dependency-plugin:2.0-alpha-1-snapshot? The code has been refactored and I believe the marker now uses the classifer too.

**Comment by jira-importer** [comment]
**[duality72](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=duality72)** commented

I tried it with the new snapshot version and it did indeed work fine. Don't you love those kinds of bugs?

I also tried the overwrite flag and even the markerFile property found in the old documentation, but got the "no setters" error for them and, looking at the Javadoc, it does indicate that there are no setters for those properties in ArtifactItem. So I'll stick with the above workaround for now.
